modules/conversantAnalyticsAdapter.js

```diff
@@ -0,0 +1,470 @@
+ *
+ * Current assumption is that the timeout will always be an array even if it is just one object in the array.
+ * @param args  [{
+  "bidId": "80882409358b8a8",
+    "bidder": "conversant",
+    "adUnitCode": "MedRect",
+    "auctionId": "afbd6e0b-e45b-46ab-87bf-c0bac0cb8881"
+  }, {
+    "bidId": "9da4c107a6f24c8",
+    "bidder": "conversant",
+    "adUnitCode": "Leaderboard",
+    "auctionId": "afbd6e0b-e45b-46ab-87bf-c0bac0cb8881"
+  }
+ ]
+ */
+function onBidTimeout(args) {
+  args.forEach(timedOutBid => {
+    const timeoutCacheKey = getLookupKey(timedOutBid.auctionId, timedOutBid.adUnitCode, timedOutBid.bidder);
+    timeoutCache[timeoutCacheKey] = {
+      timeReceived: Date.now()
+    }
+  });
+}
+
+/**
+ * Bid won occurs after auctionEnd so we need to send this separately. We also save an entry in the adIdLookup cache
+ * so that if the render fails we can match up important data so we can send a valid RENDER FAILED event back.
+ * @param args bidWon args
+ */
+function onBidWon(args) {
+  const bidderCode = args.bidderCode;
+  const adUnitCode = args.adUnitCode;
+  const auctionId = args.auctionId;
+  // Make sure we have all the data we need
+  if (typeof bidderCode === 'undefined' || typeof adUnitCode === 'undefined' || typeof auctionId === 'undefined') {
+    utils.logError(LOG_PREFIX + 'onBidWon() did not get all the necessary data to process the event.');
+    return;
+  }
+  const bidWonPayload = createPayload('bid_won', auctionId);
+
+  const adUnitPayload = createAdUnit();
+  bidWonPayload.adUnits[adUnitCode] = adUnitPayload;
+
+  const bidPayload = initializeBidDefaults();
+  bidPayload.eventCodes.push(CNVR_WIN);
+  bidPayload.adSize = createAdSize(args.width, args.height);
+  bidPayload.cpm = args.cpm;
+  bidPayload.originalCpm = args.originalCpm;
+  bidPayload.currency = args.currency;
+  bidPayload.timeToRespond = args.timeToRespond;
+  adUnitPayload.bids[bidderCode] = bidPayload;
+
+  if (typeof adIdLookup[args.adId] === 'undefined') {
+    adIdLookup[args.adId] = {
+      'bidderCode': bidderCode,
+      'adUnitCode': adUnitCode,
+      'auctionId': auctionId,
+      'timeReceived': Date.now() // For cache cleaning
+    };
+  }
+
+  sendData(bidWonPayload);
+}
+
+/**
+ * RENDER FAILED occurs after AUCTION END and BID WON, the payload does not have all the data we need so we use
+ * adIdLookup to pull data from a BID WON event to populate our payload
+ * @param args = {
+ *  reason: <value>
+ *  message: <value>
+ *  adId: <value> --optional
+ *  bid: {object?} --optional: unsure what this looks like but guessing it is {bidder: <value>, params: {object}}
+ *    }
+ */
+function onAdRenderFailed(args) {
+  const adId = args.adId;
+  // Make sure we have all the data we need, adId is optional so it's not guaranteed, without that we can't match it up
+  // to our adIdLookup data.
+  if (typeof adId === 'undefined' || typeof adIdLookup[adId] === 'undefined') {
```

Comment:
Remove typeof's.  The second one could work either way, but not needed.  Actually not even need to check against 'undefined'.  

```javascript
if (!adId || !adIdLookup[adId]) {
```